### ISSUE TITLE
SSO Admin: ListInstances permission failure when region set to "*" for the account

### ISSUE THREAD

**Issue by idesofoctober** [turbot/steampipe-plugin-aws#1879]
**Describe the bug**
When I run a query against an account that is setup with aws.spc and credentials that have the right to run "sso:ListInstances", I get this error: 

Error: operation error SSO Admin: ListInstances, https response error StatusCode: 400, RequestID: 6730cd40-2c79-45d3-b62d-0e758388b9d3, AccessDeniedException: User: arn:aws:sts::REDACTED:assumed-role/AWSReservedSSO_REDACTED_REDACTED is not authorized to perform: sso:ListInstances 

**Steampipe version (`steampipe -v`)**
Example: v0.20.6

**Plugin version (`steampipe plugin list`)**
Example: v0.5.0

**To reproduce**
Attempt to run this query from shell and/or within the steampipe "app":
select * from REDACTED.aws_ssoadmin_permission_set 

When the account prefixing the permission set table is setup in aws.spc with "regions = ["*"]

**Expected behavior**
Get results from query
**Additional context**
When updating regions to a specific region (in my case us-east-1, this succeeds)

**Comment by graza-io** [comment]
Hi @idesofoctober 

> tl;dr - can you verify that your user/role has permissions via PolicySim for all regions?

Having taken a look into this I am able to replicate the issue under various conditions, however they don't appear to be an issue with our plugin but rather that the user/role doesn't actually have the permissions across `all` regions.

The quickest way to demonstrate this would be to give a basic example:

I set-up a role which has `ReadOnlyAccess` [AWS managed policy] this includes the permission `sso:List*` within the `Allow`.

I than added a custom inline policy to `Deny` `sso:List*` unless the region is `eu-north-1` to the Role (however this was just a simple example, in reality the `Deny` here could come from an SCP, Control Tower, etc):
```json
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Sid": "TestDenySSOList",
      "Effect": "Deny",
      "Action": ["sso:List*"],
      "Resource": ["*"],
      "Condition": {
        "StringNotEquals": {
          "aws:RequestedRegion": ["eu-north-1"]
        }
      }
    }
  ]
}
```

A quick demonstration in PolicySim shows I can access fine from one region
<img width="1650" alt="eu_north_1" src="https://github.com/turbot/steampipe-plugin-aws/assets/53316123/c4555d17-c615-4c7a-be90-38bdf40db2a2">

But not another
<img width="1647" alt="us_east_1" src="https://github.com/turbot/steampipe-plugin-aws/assets/53316123/bbac7900-50cd-4b4e-a5c0-fb7aad260570">

Then when attempting to run with the following config in `aws.spc`
```hcl
connection "aws_role_a" {
  plugin  = "aws"
  profile = "graza-role-a"
  regions = ["*"]
}
```

```
> select * from aws_role_a.aws_ssoadmin_permission_set

Error: operation error SSO Admin: ListInstances, https response error StatusCode: 400, RequestID: 5fe69c0a-aa49-4d2c-93c5-28e0ff12c33c, AccessDeniedException: User: **ROLE-ARN** is not authorized to perform: sso:ListInstances (SQLSTATE HV000)

+------+-----+--------------+-------------+--------------+-------------+------------------+----------+------+-------+------+-----------+--------+------------+------+
| name | arn | created_date | description | instance_arn | relay_state | session_duration | tags_src | tags | title | akas | partition | region | account_id | _ctx |
+------+-----+--------------+-------------+--------------+-------------+------------------+----------+------+-------+------+-----------+--------+------------+------+
+------+-----+--------------+-------------+--------------+-------------+------------------+----------+------+-------+------+-----------+--------+------------+------+
```

Updating my configuration:
```hcl
connection "aws_role_a" {
  plugin  = "aws"
  profile = "graza-role-a"
  regions = ["eu-north-1"]
}
```

```
> select * from aws_role_a.aws_ssoadmin_permission_set
+------+-----+--------------+-------------+--------------+-------------+------------------+----------+------+-------+------+-----------+--------+------------+------+
| name | arn | created_date | description | instance_arn | relay_state | session_duration | tags_src | tags | title | akas | partition | region | account_id | _ctx |
+------+-----+--------------+-------------+--------------+-------------+------------------+----------+------+-------+------+-----------+--------+------------+------+
+------+-----+--------------+-------------+--------------+-------------+------------------+----------+------+-------+------+-----------+--------+------------+------+
```

**Comment by graza-io** [comment]
Additional note: you can make use of wildcards in your regions so `["us-*","eu-*"]` would be valid for all regions in the US and EU.

**Comment by graza-io** [comment]
Further Notes:

It seems that you potentially need `SSO` to be enabled in region you're querying against also.

You could add `AccessDeniedException` to your `aws.spc` configuration file, to ignore these errors and allow the plugin to continue:
`ignore_error_codes = ["AccessDeniedException"]`

So for example:
```hcl
connection "aws_xyz" {
  plugin  = "aws"
  profile = "xyz"
  regions = ["*"]
  ignore_error_codes = ["AccessDeniedException"]
}
```

Alternatively you could also have a dedicated connection for querying your SSO admin tables which only specifies one region:
```hcl
connection "aws_sso" {
  plugin  = "aws"
  profile = "xyz"
  regions = ["us-east-1"]
}

connection "aws_xyz" {
  plugin  = "aws"
  profile = "xyz"
  regions = ["*"]
}
```

Then when querying and needing to join data you would prefix the SSO Admin tables with `aws_sso.aws_ssoadmin_permission_set` and other tables with `aws_xyz`.

**Comment by graza-io** [comment]
Closing as advice issued.
